pre_commit/meta_hooks/check_files_matches_any.py

```diff
@@ -0,0 +1,36 @@
+import re
```

Comment:
My thoughts with this hook (again, I didn't communicate this, sorry!) is to make `(no files to check)Skipped` an *error* when running with `--all-files`.  So what I'd like to see in this hook is a reuse of the `include` / `exclude` for both files and types from `pre_commit.commands.run` -- the messaging can be `"{hook_id} does not apply to this repository"` or something of the sort